tests/integration/declare_txn_tests.rs

```diff
@@ -0,0 +1,68 @@
+use blockifier::context::BlockContext;
+use blockifier::declare_tx_args;
+use blockifier::execution::contract_class::ClassInfo;
+use blockifier::test_utils::NonceManager;
+use blockifier::transaction::test_utils::max_fee;
+use rstest::rstest;
+use snos::crypto::poseidon::PoseidonHash;
+use snos::starknet::business_logic::utils::write_class_facts;
+use snos::storage::storage_utils::compiled_contract_class_cl2vm;
+use starknet_api::core::CompiledClassHash;
+use starknet_api::transaction::{Fee, TransactionVersion};
+
+use crate::common::block_context;
+use crate::common::state::{initial_state_cairo1, load_cairo1_contract, StarknetTestState};
+use crate::common::transaction_utils::execute_txs_and_run_os;
+
+#[rstest]
+// We need to use the multi_thread runtime to use task::block_in_place for sync -> async calls.
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn declare_cairo1_account(
+    #[future] initial_state_cairo1: StarknetTestState,
+    block_context: BlockContext,
+    max_fee: Fee,
+) {
+    let initial_state = initial_state_cairo1.await;
+
+    let tx_version = TransactionVersion::TWO;
+    let mut nonce_manager = NonceManager::default();
+
+    let account_contract = initial_state.cairo1_contracts.get("account_with_dummy_validate").unwrap();
+
+    // we want to declare a fresh (never-before-declared) contract, so we don't want to reuse
+    // anything from the test fixtures, and we need to do it "by hand"
+    let (_, sierra_class, casm_class) = load_cairo1_contract("empty_contract");
+    let mut ffc = initial_state.clone_ffc::<PoseidonHash>();
+    let (contract_class_hash, compiled_class_hash) =
```

Comment:
Still missing a comment here, could be more explicit that we need to store the contract in the storage ahead of the tx.

Reply:
ccd4de9